projects/js-packages/components/components/text/types.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+import React, { CSSProperties } from 'react';
+
+import { VARIANTS_MAPPING } from './constants';
+
+export type SpacingValue = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8;
+
+export type TextProps = {
+	variant?: keyof typeof VARIANTS_MAPPING;
+	/** margin */
+	m?: SpacingValue;
+	/** margin-top */
+	mt?: SpacingValue;
+	/** margin-right */
+	mr?: SpacingValue;
+	/** margin-bottom */
+	mb?: SpacingValue;
+	/** margin-left */
+	ml?: SpacingValue;
+	/** margin left and right */
+	mx?: SpacingValue;
+	/** margin top and bottom */
+	my?: SpacingValue;
+	/** padding */
+	p?: SpacingValue;
+	/** padding-top */
+	pt?: SpacingValue;
+	/** padding-right */
+	pr?: SpacingValue;
+	/** padding-bottom */
+	pb?: SpacingValue;
+	/** padding-left */
+	pl?: SpacingValue;
+	/** padding left and right */
+	px?: SpacingValue;
+	/** padding top and bottom */
+	py?: SpacingValue;
+	/** HTML Class */
+	className?: string;
+	/** The text itself that will be rendered. */
+	children?: React.ReactNode;
+	/** Force an specific tag (span, div) or use a custom component that will receive className and children */
+	component?: React.FC< Omit< TextProps, 'variant' | 'component' > > | React.ElementType;
+};
+
+export type H3Props = TextProps & {
+	/** Font weight: 'bold' (default) | 'regular'. */
+	weight?: CSSProperties[ 'fontWeight' ];
```

Comment:
<3

Reply:
actually, what are the possible values that `CSSProperties[ 'fontWeight' ]` provides? I bet it's possible to define `400`, for instance, which is not expected by the `<H3 />` component.

Reply:
Yes, it's possible to pass numbers and other string values.

After checking the styles again, I think only `bold` and `regular` is supported/expected.

Reply:
Fixed, thanks